src/Interpreters/SystemLog.h

```diff
@@ -229,9 +230,18 @@ void SystemLog<LogElement>::startup()
 }
 
 
+static thread_local bool recursive_add_call = false;
+
 template <typename LogElement>
 void SystemLog<LogElement>::add(const LogElement & element)
 {
+    /// It is possible that the method will be called recursively.
```

Comment:
But how is it possible? Inner logging inside text_log itself? This comment doesn't help at all.

Reply:
std::vector::push_back -> operator new -> MemoryTracker::alloc -> MemoryTracker::logPeakMemoryUsage -> logger -> SystemLog<TextLog>::add